internal/release/promote/promotion.go

```diff
@@ -131,20 +133,23 @@ func (p *Promotion) Promote(opts Opts) (string, error) {
 	if err != nil {
 		return "", fmt.Errorf("selecting releases to promote: %w", err)
 	}
-	if !list.HasAnyPromotableReleases() {
-		p.promptProvider.PrintNoPromotableReleasesFound(opts.ReleasesFiltered, opts.SourceEnv, opts.TargetEnv)
-		return "", nil
-	}
 
 	if !opts.NoPrompt {
-		if err := p.preview(list); err != nil {
+		if err := p.preview(selectedList); err != nil {
```

Comment:
Is the preview the `git diff`? If so it might worthwhile to error out before showing the diff.